tests/unit/moduleapi/datatype.tcl

```diff
@@ -89,4 +89,20 @@ start_server {tags {"modules"}} {
         $rd read
         $rd close
     }
+    test "SCAN module datatype" {
+        r flushdb
+        r datatype.set foo 111 bar
```

Comment:
let's add another key of a different type, so that the filtering is effective.
also, please check if the TYPE command has a test already, and if not, maybe add an assertion below.

Reply:
don't forget about the suggestion to avoid sdsdup which i mistakenly posted in the wrong thread:
https://github.com/redis/redis/pull/12209#discussion_r1233546039